chromadb/server/fastapi/__init__.py

```diff
@@ -985,7 +1792,7 @@ async def reset(
         )
 
     @trace_method("FastAPI.get_nearest_neighbors", OpenTelemetryGranularity.OPERATION)
-    async def get_nearest_neighbors(
+    async def get_nearest_neighbors_v1(
```

Comment:
It's not the most straightforward to do so and includes module-ing FastAPI. I think it should be somewhat more readable now with the split into functions. If it's still tough to read, lmk and I'll go down that route but felt inefficient given that we're immediately deleting the v1 routes.

Reply:
Fair but, Isn't it just a matter of creating a wrapper class that extends this and moving the functions there?

Reply:
I don't believe it's that straightforward—wouldn't that create two running instances of the server?

Reply:
No, you only instantiate one.

Server() -> Say this is current class
ServerV2(Server) -> Adds v2 routes

or alternatively
Server -> Current class with v2 routes only
ServerV1(Server) -> Adds v1 routes for bw compat.

This is fine as is, but I think it should be simple

Reply:
Ah I see what you mean. Updated to wrap with class that has v1 routes

Reply:
Decided to revert this and keep all in one file because it will break hosted Chroma otherwise. This is how I implemented your suggestion:
```
class FastAPI(Server):
# Setup v2 routes

class FastAPIWithV1(FastAPI):
# Setup v1 routes
# Post route setup
self._app.include_router(self.router)
use_route_names_as_operation_ids(self._app)
... 

# In chromadb/app.py:
from chromadb.server.fastapi.v1 import FastAPIWithV1
server = FastAPIWithV1(settings)
```
There is some configuration that necessarily has to happen post adding routes, so we can only fire it in the wrapper class. We thus also have to update the app to use the wrapper class. This makes it such that hosted Chroma will fail once we deploy changes, since it initializes the server in `python/chroma_hosted/chroma_hosted/app.py`. It won't have access to the wrapper class, which has the necessary post route configuration. 

There is a way around this, by doing something like:
```
class FastAPI(Server):
...
class FastAPI(FastAPIWithV1):
...
```
However, in this case we still have to split things up in an awkward way, like moving router initialization and it doesn't make for a clean delete of the old routes, especially if there's some time between and context is lost. Overall, I think it's cleaner to keep things all in one file.